src/aleph/web/controllers/storage.py

```diff
@@ -118,18 +108,53 @@ def content(self) -> Union[str, bytes]:
 
 class MultipartUploadedFile:
     _content: Optional[bytes]
+    metadata: bool
+    size = 0
 
-    def __init__(self, file_field: FileField):
+    def __init__(self, file_field: FileField, metadata: bool):
+        self.metadata = metadata
         self.file_field = file_field
+        self.MAX_SIZE = (
+            MAX_UPLOAD_FILE_SIZE
+            if self.metadata
+            else MAX_UNAUTHENTICATED_UPLOAD_FILE_SIZE
+        )
 
         try:
-            content_length_str = file_field.headers["Content-Length"]
-            self.size = int(content_length_str)
+            if self.metadata:
+                content_length_str = file_field.headers["Content-Length"]
+                self.size = int(content_length_str)
+                self._content = self.file_field.file.read()
+            else:
+                self._content = self.read_file_with_max_size(
+                    MAX_UNAUTHENTICATED_UPLOAD_FILE_SIZE
+                )
+                if self._content:
+                    self.size = len(self._content)
         except (KeyError, ValueError):
             raise web.HTTPUnprocessableEntity(
                 reason="Invalid/missing Content-Length header."
             )
-        self._content = None
+
+    def read_file_with_max_size(self, max_size: int) -> Union[bytes, None]:
+        buffer_size = 64 * 1024  # 64 KB buffer size
+
+        content = b""
+        total_read = 0
+
+        while True:
+            chunk = self.file_field.file.read(buffer_size)
+
+            if not chunk:
+                break
+
+            total_read += len(chunk)
+            if total_read > max_size:
+                raise web.HTTPForbidden()
```

Comment:
Throw `413 Payload Too Large` instead. Throw `400 Bad Request` if Content-Length does not match actual body bytes.